src/main/kotlin/org/ptss/support/security/AuthenticationFilter.kt

```diff
@@ -0,0 +1,79 @@
+package org.ptss.support.security
+
+
+import io.quarkus.security.UnauthorizedException
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerRequestFilter
+import jakarta.ws.rs.container.ResourceInfo
+import jakarta.ws.rs.core.Context
+import jakarta.ws.rs.core.HttpHeaders
+import jakarta.ws.rs.core.NewCookie
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+import org.eclipse.microprofile.config.inject.ConfigProperty
+
+
+@Provider
+@ApplicationScoped
+class AuthenticationFilter @Inject constructor(
+    @Context private val resourceInfo: ResourceInfo,
+    private val identityServiceClient: IdentityServiceClient,
+    private val jwtValidator: JwtValidator,
+    @ConfigProperty(name = "app.security.access-token-cookie-name", defaultValue = "access_token")
+    private val accessTokenCookieName: String,
+    @ConfigProperty(name = "app.security.refresh-token-cookie-name", defaultValue = "refresh_token")
+    private val refreshTokenCookieName: String
```

Comment:
These should come from your requests right?  Why do they come from application.properties instead?
would be fine if you did this, no?
`val accessToken = requestContext.cookies[accessTokenCookieName]?.value`

Reply:
What do you mean come from requests, how can you know from a request what the name of the refresh token and access token is?